posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,560 @@
+description: This blog explores a beautiful relationship between Serverless and GraphQL.  
+date: 2018-02-13  
+layout: Post  
+thumbnail: https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/graphql.jpeg  
+authors:  
+  - SiddharthGupta
+---
+
+![alt text](https://user-images.githubusercontent.com/1587005/36030132-d45feefe-0d5a-11e8-986c-92dad62f3408.png "Serverless and GraphQL family")
+
+## Introduction
+
+Over the last four years, I've been exploring the world of big data, building real-time and batch systems at scale. For the last couple of months, I've been building products with serverless architectures here at Glassdoor.
+
+Given the intersection here of serverless and big data, here are few questions on everyone's mind: 
+
+1) How can we build low latency APIs to server these complex, high dimensional and big datasets which can scale on demand?
+3) Can we fire single query which can construct a nested response from multiple data sources?
+4) Is there an easy way to secure the endpoint, paginate through the data, aggregate the results at scale and what not? 
+5) Can we pay per query execution rather than running fleet of servers?
+
+The answer? GraphQL.
+
+> “Engineers like to solve problems. If there are no problems handily available, they will create their own problems.” - Scott Adams
+
+I am sure you have heard it before. You might even be, unknowingly, creating more problems than solutions in your application.
+
+This blog aims to show you one key way you can streamline lots of your work. Namely, the beautiful relationship between Serverless and GraphQL.
+
+In this post, I’ll cover how to create Serverless GraphQL endpoints using DynamoDB and RDS; we'll wrap it all around a REST API.
+
+## What is GraphQL?
+
+I’m going to start this off with a statement of truth, to all my fellow engineers: The way we currently build APIs, where all of them are split up and maintained separately, isn’t optimal.
+
+Luckily for us, the tech horizon is ever-expanding. We have options. And we should use them.
+
+[GraphQL](https://dev-blog.apollodata.com/2017-the-year-in-graphql-124a050d04c6) lets you shrink your multitude of APIs down into a single HTTP endpoint. In short, it provides a simple way of building mobile/web applications by providing a clean layer of abstraction between servers and clients.
+
+It lets you:
+1. Use a single endpoint to fetch data from multiple data sources (meaning reduced network costs and better query efficiency).
+2. Know exactly what your response will look like. Ensure you're never sending more or less than the client needs.
+3. Describe your API with types that map your schema to existing backend.
+
+Thousands of companies are now using GraphQL in production with the help of open source frameworks built by Facebook, Apollo, and Graphcool. In fact, that [Starbucks announcement](https://twitter.com/davidbrunelle/status/960946257643454464) last week is going to make my morning coffee taste even better. 😉
```

Comment:
we need to fix this. I don't like startbucks tweet on more than half of the page.